test/unit/compare.hpp

```diff
@@ -0,0 +1,31 @@
+
+template<typename T1, typename T2, typename S1, typename S2>
+void compare(const std::map<T1,T2> &a, const std::map<S1,S2> &b) {
+  ASSERT_EQ(a.size(), b.size());
+  for(auto const& [key, value] : a)
+    EXPECT_EQ(b.at(key), value);
+}
+
+#ifdef _BOOST_UBLAS_MATRIX_
+template<typename T1, typename T2>
+void compare(const ublas::matrix<T1> &m1, const ublas::matrix<T2> &m2) {
+    ASSERT_EQ(m1.size1(), m2.size1());
+    ASSERT_EQ(m1.size2(), m2.size2());
+    for(int i = 0; i < m1.size1(); i++)
+        for(int j = 0; j < m1.size2(); j++)
+            EXPECT_EQ(m1(i, j), m2(i, j));
```

Comment:
morda raje EXPECT_LT(std::abs(m1(i,j)-m2(i,j)), epsilon)
epsilon pa naj bo defaulted parameter funkcije, recimo 1e-10

Reply:
gtest ima EXPECT_DOUBLE _EQ in EXPECT_FLOAT_EQ, ki sta precej bolj pametna se mi zdi, primerja z ULPi kot bi blo treba dejansko, bom zamenjal

Reply:
Aha, saj res! Ja, to je se boljsa resitev!